### ISSUE TITLE
Bug: flaky test checks log is empty, but esm_cache service is running in the background

### ISSUE THREAD

**Issue by panlinux** [canonical/ubuntu-pro-client#3001]
**Description of the bug**
`Scenario Outline: Non-root user and root user log files are different` is a flaky test because while it truncates the log file, the esm_cache service is running in the background and populating the log again.

```
2024-03-18T22:04:06.8536613Z   Scenario Outline: Non-root user and root user log files are different -- @1.1 ubuntu release  # features/logs.feature:55
2024-03-18T22:04:06.8538826Z     Given a `xenial` `lxd-container` machine with ubuntu-advantage-tools installed              # features/steps/machines.py:161
2024-03-18T22:04:18.6790000Z     When I run `truncate -s 0 /var/log/ubuntu-advantage.log` with sudo                          # features/steps/shell.py:54
2024-03-18T22:04:18.6907151Z     And I verify `/var/log/ubuntu-advantage.log` is empty                                       # features/steps/files.py:39
2024-03-18T22:04:19.9291948Z       Assertion Failed:
2024-03-18T22:04:19.9293041Z       Expected: a string matching '0'
2024-03-18T22:04:19.9293895Z            but: was '1'
```

And the log has:
```
["2024-03-18T22:04:18.727", "DEBUG", "ubuntupro.apt", "done", 891, "Fetched ESM Apt Cache item: gpgv:/var/lib/ubuntu-advantage/apt-esm/var/lib/apt/lists/partial/esm.ubuntu.com_apps_ubuntu_dists_xenial-apps-security_InRelease", {}]
["2024-03-18T22:04:18.764", "DEBUG", "ubuntupro.apt", "done", 891, "Fetched ESM Apt Cache item: gpgv:/var/lib/ubuntu-advantage/apt-esm/var/lib/apt/lists/partial/esm.ubuntu.com_apps_ubuntu_dists_xenial-apps-security_InRelease", {}]
["2024-03-18T22:04:18.814", "DEBUG", "ubuntupro.apt", "done", 891, "Fetched ESM Apt Cache item: gpgv:/var/lib/ubuntu-advantage/apt-esm/var/lib/apt/lists/partial/esm.ubuntu.com_apps_ubuntu_dists_xenial-apps-updates_InRelease", {}]
["2024-03-18T22:04:18.839", "DEBUG", "ubuntupro.apt", "done", 891, "Fetched ESM Apt Cache item: gpgv:/var/lib/ubuntu-advantage/apt-esm/var/lib/apt/lists/partial/esm.ubuntu.com_apps_ubuntu_dists_xenial-apps-updates_InRelease", {}]
["2024-03-18T22:04:18.906", "DEBUG", "ubuntupro.apt", "done", 891, "Fetched ESM Apt Cache item: gpgv:/var/lib/ubuntu-advantage/apt-esm/var/lib/apt/lists/partial/esm.ubuntu.com_infra_ubuntu_dists_xenial-infra-security_InRelease", {}]
["2024-03-18T22:04:18.930", "DEBUG", "ubuntupro.apt", "done", 891, "Fetched ESM Apt Cache item: gpgv:/var/lib/ubuntu-advantage/apt-esm/var/lib/apt/lists/partial/esm.ubuntu.com_infra_ubuntu_dists_xenial-infra-security_InRelease", {}]
["2024-03-18T22:04:18.966", "DEBUG", "ubuntupro.config", "parse_config", 622, "Using client configuration file at /etc/ubuntu-advantage/uaclient.conf", {}]
```

The `truncate` happened at 22:04:18.6790000Z, but we see the esm cache service still fetching data after that.

The test should either sleep (ugh), poll esm cache until it finishes, or perhaps stop that service before running.

**Expected behavior**

Non-flaky tests, we like them

**Current behavior**
Flaky test

**To Reproduce**
Run behave a few times

**System information:**

- Ubuntu release: xenial behave system
- Pro Client version: main branch

**Comment by renanrodrigo** [comment]
definitely stop all services is the way to go - we have been bothered by those flakies more than we would like really. Thanks for reporting

**Comment by renanrodrigo** [comment]
@orndorffgrant do you think we can live with the error until the journald PR gets merged? If we patch the test now we need to unpatch it later

**Comment by orndorffgrant** [comment]
Just merged #2779 which fixes this
